public/src/components/epicTests/epicTestEditor.tsx

```diff
@@ -29,43 +33,93 @@ const styles = ({ palette, spacing, mixins }: Theme) => createStyles({
   variantListHeading: {
     fontWeight: "bold"
   },
-  userCohortSelect: {
-    width: "50%"
+  formControl: {
+    marginTop: spacing.unit * 2,
+    marginBottom: spacing.unit,
+    minWidth: "60%",
+    maxWidth: "100%",
+    display: "block",
   }
 });
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
 
+type EpicTestVariantsState = {
+  selectedVariantName?: string
+}
 interface Props extends WithStyles<typeof styles> {
   test?: EpicTest,
   onChange: (updatedTest: EpicTest) => void
 }
 
-class EpicTestsEditor extends React.Component<Props, any> {
+enum TestFieldNames {
+  locations = "locations",
+  tagIds = "tagIds",
+  sections = "sections",
+  excludedTagIds = "excludedTagIds",
+  excludedSections = "excludedSections",
+  alwaysAsk = "alwaysAsk",
+  userCohort = "userCohort",
+  isLiveBlog = "isLiveBlog",
+  hasCountryName = "hasCountryName",
+  variants = "variants"
+}
+class EpicTestEditor extends React.Component<Props, any> {
 
+  state: EpicTestVariantsState = { selectedVariantName: undefined}
 
-  onListChange = (fieldName: string) => (updatedString: string): void => {
-    if (this.props.test) {
-      const updatedTest = {
-        ...this.props.test,
-        [fieldName]: updatedString.split(",")
-      };
-      this.props.onChange(updatedTest)
-    }
+  onVariantSelected = (variantName: string): void => {
+    this.setState({
+      selectedVariantName: variantName
+    })
   };
 
-  onUserCohortChange = (cohort: UserCohort) => {
+
+  updateTest = (fieldName: TestFieldNames, updatedData: string | (string | EpicVariant)[] | boolean) => {
```

Comment:
I know I started this with `onListChange`, but looking at this again I think we should avoid this pattern.
The function signature gives the illusion of type safety, but actually the compiler cannot guarantee that the type of `updatedData` matches the `fieldName`. E.g. you could call `updateTest('isLiveBlog', 'should be a bool')` and it will not complain.

How about this instead? Not as clean, but it type checks:
```
  updateTest = (update: (test: EpicTest) => EpicTest) => {
    if (this.props.test) {
      this.props.onChange(update(this.props.test))
    }
  };

  onUserCohortChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    let selectedCohort = event.target.value as UserCohort;
    this.updateTest(test => ({ ...test, "userCohort": selectedCohort}));
  };
```

Reply:
Great - I think I've updated both the updateTest and updateVariant accordingly now.